plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/PrimitiveTypeMapBuilder.java

```diff
@@ -42,7 +43,7 @@ private PrimitiveTypeMapBuilder() {}
     private Map<List<String>, Type> buildTypeMap(List<Type> types, List<String> columnNames)
     {
         for (int i = 0; i < types.size(); i++) {
-            visitType(types.get(i), columnNames.get(i), ImmutableList.of());
+            visitType(types.get(i), AvroSchemaUtil.makeCompatibleName(columnNames.get(i)), ImmutableList.of());
```

Comment:
Does `visitRowType` also need this?

Reply:
Do we have a test for ROW types with fancy column names? I remember something was added for ORC and case-sensitivity in https://github.com/trinodb/trino/pull/7350/files for example.

Reply:
@hashhar visitRowType doesn't need that.